### PR TITLE
Update to GCC 11

### PR DESCRIPTION
**Pull request by bdice**
This PR updates builds to use GCC 11.

### CODE REVIEW

**Comment by bdice** [comment]
This is ready for review, but please **_do not merge_** until all RAPIDS repos are ready to migrate. (I do not have privileges to add the DO NOT MERGE label to prohibit merging.)

**Review by jjacobelli — APPROVED** [review]

File: dependencies.yaml [hunk 1]
```diff
@@ -59,12 +59,12 @@ dependencies:
...
               arch: x86_64
             packages:
-              - gcc_linux-64=9
+              - gcc_linux-64=11.*
               - sysroot_linux-64==2.17
           - matrix:
               arch: aarch64
             packages:
-              - gcc_linux-aarch64=9
+              - gcc_linux-aarch64=11.*
```
Comment (jjacobelli) [inline dependencies.yaml:67]: Approved, but I'm curious, what's the difference between `=11` and = `11.*` here?

Comment (bdice) [reply dependencies.yaml:67]: I'm aligning with the way that we specified this in other packages (cudf, cuspatial, cusignal, ...). I'm not sure which to prefer, I think both may be equivalent.

https://github.com/rapidsai/cudf/blob/b6d9fd1f26ccda9b3043a8967621bae234187c37/dependencies.yaml#L80

Ref: https://docs.conda.io/projects/conda-build/en/latest/resources/package-spec.html#examples

**Review by cjnolet — APPROVED** [review]
LGTM

**Pull request merged**